app/queries/answers.ts

```
@@ -0,0 +1,51 @@
+import prisma from "@/app/services/prisma";
+
+export async function getQuestionsNeedingCorrectAnswer() {
+  return await prisma.$queryRaw<
+    {
+      id: number;
+      answerCount: number;
+      revealAtDate: number;
+      revealAtAnswerCount: number;
+    }[]
+  >`
+    SELECT
+      "id",
+      "answerCount",
+      "revealAtDate",
+      "revealAtAnswerCount"
+    FROM (
+      SELECT
+        q.id,
+        "revealAtDate",
+        "revealAtAnswerCount",
+        (
+          SELECT
+            COUNT(DISTINCT CONCAT(qa."userId", qo."questionId"))
+          FROM
+            "QuestionOption" qo
+          JOIN
+            "QuestionAnswer" qa ON qa."questionOptionId" = qo."id"
+          WHERE
+            qo."questionId" = q."id"
+      ) AS "answerCount"
+      FROM
+        "Question" q
+      LEFT JOIN
+        "QuestionOption" qo ON q.id = qo."questionId"
+      LEFT JOIN
+        "QuestionAnswer" qa ON qo.id = qa."questionOptionId"
+      WHERE
+        ("calculatedIsCorrect" IS NULL OR "calculatedAveragePercentage" IS NULL)
+      GROUP BY q.id
+    )
+    WHERE (
+      (
+        ("revealAtDate" IS NOT NULL AND "revealAtDate" < NOW())
+        OR ("revealAtAnswerCount" IS NOT NULL AND "revealAtAnswerCount" >= "answerCount")
+      ) AND (
+        "answerCount" >= ${Number(process.env.MINIMAL_ANSWERS_PER_QUESTION ?? 3)}
```

I think if env doesn't exist we should through an error may be. What do you think?

---

I don't think we should throw a runtime error due to missing config.

I'm not a fan of having default values scattered throughout the code like this but this pattern is already established.

---

Can you change this so that we can eventually remove this pattern?

---

I changed it for now. The correct, long-term solution is to check the config on startup.